### PR TITLE
Random rooms now load at roundstart

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

Based on #5443 with my edits but the code is mostly the same
## Why It's Good For The Game
We don't have to rely on timers

All the credit goes to jupyterkat

## Changelog 
:cl: jupyterkat
add: Random rooms are initialized at the start of the round.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
the names of the people you want in the CL need to be after :cl: rather than in the header of the section, and if it involves jupiterkat is this tested?

**Comment by Archanial** [comment]
Yes, loaded meta and rooms spawned.

File: code/controllers/subsystem/mapping.dm [hunk 1]
```diff
@@ -228,15 +229,39 @@ SUBSYSTEM_DEF(mapping)
...
+		if(possibletemplates.len)
+			R.template = pickweight(possibletemplates)
+			R.template.stock--
+			R.template.weight = (R.template.weight / 2)
+			if(R.template.stock <= 0)
+				R.template.spawned = TRUE
+			R.template.stationinitload(get_turf(R), centered = R.template.centerspawner)
+		qdel(R)
+	//We don't need those anymore
+	random_room_templates = list()
```
Comment (Archanial) [inline code/controllers/subsystem/mapping.dm]:
```suggestion
	random_room_templates = null
```

**Comment by zeskorion** [comment]
Randomroom spawners can be created in space ruins. This also removes my ability to make nested randomrooms or include randomrooms in randommaint v2

**Comment by Archanial** [comment]
I realized the global list is stupid
draft until I rework it to not be stupid

**Comment by PowerfulBacon** [comment]
> Randomroom spawners can be created in space ruins. This also removes my ability to make nested randomrooms or include randomrooms in randommaint v2

good point, these can spawn after mapload

**Comment by Archanial** [comment]
They can and will, read the code.

**Pull request merged**